lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -51,12 +51,6 @@
         description:
             - Name of the newly deployed guest
         required: True
-   name_match:
-        description:
-            - If multiple vms matching the name, use the first or last found
-        required: False
-        default: 'first'
-        choices: ['first', 'last']
```

Comment:
I agree it is not good practice, but sadly vmware does not enforce this and people saw a need for this option. As for ANY existing option, it should not be removed arbitrarily as it breaks backwards compatiblity